src/blocks/remote-data-template/components/loop-template/LoopTemplate.tsx

```diff
@@ -30,8 +29,10 @@ export function LoopTemplate( props: LoopTemplateProps ) {
 		[ clientId ]
 	);
 
+	// Don't render anything if there are no results.
+	// Leave it to the no results block to handle this.
 	if ( ! remoteData.results.length ) {
-		return <p>{ __( 'No results found.' ) }</p>;
+		return null;
```

Comment:
This stops double "no results found" by ensuring that only the no results block handles this case